numba/tests/test_mathlib.py

```diff
@@ -518,14 +519,38 @@ def test_isfinite_npm(self):
 
     def test_hypot(self, flags=enable_pyobj_flags):
         pyfunc = hypot
-        x_types = [types.int16, types.int32, types.int64,
-                   types.uint16, types.uint32, types.uint64,
+        x_types = [types.int64, types.uint64,
                    types.float32, types.float64]
         x_values = [1, 2, 3, 4, 5, 6, .21, .34]
         y_values = [x + 2 for x in x_values]
         # Issue #563: precision issues with math.hypot() under Windows.
         prec = 'single' if sys.platform == 'win32' else 'exact'
-        self.run_binary(pyfunc, x_types, x_values, y_values, prec=prec)
+        self.run_binary(pyfunc, x_types, x_values, y_values, flags, prec)
+        # Check that values that overflow in naive implementations do not
+        # in the numba impl
+
+        def naive_hypot(x, y):
+            return math.sqrt(x * x + y * y)
+        for fltty in (types.float32, types.float64):
+            cr = self.ccache.compile(pyfunc, (fltty, fltty), flags=flags)
+            cfunc = cr.entry_point
+            dt = numpy_support.as_dtype(fltty).type
+            val = dt(np.finfo(dt).max / 10.)
+            nb_ans = cfunc(val, val)
+            self.assertPreciseEqual(nb_ans, pyfunc(val, val), prec='single')
+            self.assertTrue(np.isfinite(nb_ans))
+
+            # select regex based assertion function (name changed in 2->3).
```

Comment:
That's what I was going for to capture the warning caused by the overflow. Do you have a preferred method for dealing with this?